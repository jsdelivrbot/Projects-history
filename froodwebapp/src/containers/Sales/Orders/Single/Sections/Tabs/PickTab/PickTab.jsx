import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import ConnectedPickTabSection from './PickTabSection';

const mapStateToProps = state => ({
  orderId: state.order.initialValues.id,
  data: state.order.pickData,
});

export const PickTab = ({ data, orderId }) => (
  <div>
    { data.map((pickData, pickDataIndex) => (
      <ConnectedPickTabSection
        key={ pickDataIndex }
        data={ pickData }
        orderId={ orderId }
      />
    ))}
  </div>
);

PickTab.propTypes = {
  data: PropTypes.array,
  orderId: PropTypes.string.isRequired
};

export default connect(mapStateToProps)(PickTab);
