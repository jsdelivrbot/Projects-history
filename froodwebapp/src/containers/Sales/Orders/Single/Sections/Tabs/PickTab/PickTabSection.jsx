import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Row, Col } from 'react-flexbox-grid';
import { orderUpdatePickDataRequest } from 'redux-base/actions';
import { GridTable, Controls, OrderTabHeader } from 'components';

const mapDispatchToProps = { orderUpdatePickDataRequest };

const columns = [{
  title: 'Pick List #',
  dataIndex: 'code'
}, {
  title: 'Item Name',
  dataIndex: 'name'
}, {
  title: 'UOM',
  dataIndex: 'uomName'
}, {
  title: 'Order Qty',
  dataIndex: 'orderQty'
}, {
  title: 'Batch',
  dataIndex: 'batch'
}, {
  title: 'Expiry',
  dataIndex: 'expiryDate'
}, {
  title: 'Pick Qty',
  dataIndex: 'pickQty',
  type: 'number',
  min: 0
}, {
  title: 'Pick Status',
  dataIndex: 'isPicked',
  type: 'checkbox'
}];

export class PickTabSection extends Component {
  constructor(props) {
    super(props);
    this.state = {
      data: props.data.list,
      location: props.data.location
    };
  }

  componentWillReceiveProps(nextProps) {
    this.setState({
      data: nextProps.data.list,
      location: nextProps.data.location
    });
  }

  handleUpdateTableData = (data) => {
    this.setState({
      data
    });
  }

  handleUpdatePickData = () => {
    this.props.orderUpdatePickDataRequest({
      id: this.props.orderId,
      payload: {
        id: this.state.data[0].id,
        code: this.state.data[0].code,
        children: this.state.data[0].children.map(item => ({
          lineNo: item.lineNo,
          pickQty: Number(item.pickQty)
        }))
      }
    });
  }

  render() {

    const { data, location } = this.state;

    return (
      <div>
        <OrderTabHeader
          header={ location }
          headerLabel="Zone"
          primaryButtonText="Assign"
          secondaryButtonText="Print List"
        />
        <Row>
          <Col lg>
            <GridTable
              isExpandable
              disabled={ this.props.data.list[0] && this.props.data.list[0].isPicked }
              columns={ columns }
              dataSource={ data }
              updateTableData={ this.handleUpdateTableData }
            />
          </Col>
        </Row>
        <Controls
          saveButtonVisible={ this.props.data.list[0] && !this.props.data.list[0].isPicked } // items can not be picked twice
          onSaveButtonClick={ this.handleUpdatePickData }
          saveButtonText="Save"
        />
      </div>
    );
  }
}

PickTabSection.propTypes = {
  data: PropTypes.object,
  orderUpdatePickDataRequest: PropTypes.func.isRequired,
  orderId: PropTypes.string.isRequired
};

export default connect(null, mapDispatchToProps)(PickTabSection);

