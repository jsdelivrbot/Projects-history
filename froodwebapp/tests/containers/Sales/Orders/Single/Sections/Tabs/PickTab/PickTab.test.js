import { PickTab } from 'containers/Sales/Orders/Single/Sections/Tabs/PickTab/PickTab';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const { describe, expect, it } = global;

describe('PickTab', () => {
  const props = {
    data: [],
    orderNo: '1'
  };

  it('renders the PickTab component', () => {
    const component = shallow(<PickTab { ...props } />);

    expect(component.length).toEqual(1);
  });

  it('capturing Snapshot of PickTab', () => {
    const component = shallow(<PickTab { ...props } />);
    const view = toJson(component);
    const actual = expect(view);

    actual.toMatchSnapshot();
  });
});
