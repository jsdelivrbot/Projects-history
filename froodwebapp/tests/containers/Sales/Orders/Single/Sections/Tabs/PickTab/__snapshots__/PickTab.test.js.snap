// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickTab capturing Snapshot of PickTab 1`] = `<div />`;
